File: frontend/src/types/garden.ts
export interface Plant {
    id: string;
    name: string;
    position: { x: number; y: number };
}

export interface Plot {
    id: string;
    x: number;
    y: number;
    width: number;
    height: number;
    plants: string[] | number[];
}

export interface Garden {
    id: string;
    name: string;
    x: number;
    y: number;
    plots: Plot[];
    dimensions: { width: number; height: number };
}

File: backend/schemas/garden.py
from pydantic import BaseModel, Field
from typing import Optional

class Garden(BaseModel):
    name: str
    rows: int = Field(gt=0, description="Number of rows in the grid")
    cols: int = Field(gt=0, description="Number of columns in the grid")
    plants: list[Plant] = Field(default_factory=list)
